### PR TITLE
T ligu/add six a11y rules

### PR DESCRIPTION
**Pull request by t-ligu**
Add six a11y roles and tests.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 var _ = require('underscore');
 
-module.exports = function(grunt) {
+module.exports = function (grunt) {
```
Comment (ipip2005) [inline Gruntfile.js]: remove changes like this, don't change such things in existing code.

File: Gruntfile.js [hunk 2]
```diff
@@ -293,26 +298,26 @@ module.exports = function(grunt) {
...
-        var packageJson = grunt.file.readJSON('package.json', { encoding: 'UTF-8' });
-        getAllRuleNames({ skipTsLintRules: true }).forEach(function(ruleName) {
-            if (readmeText.indexOf(ruleName) === -1) {
-                grunt.fail.warn('A rule was found that is not documented in README.md: ' + ruleName);
-            }
-        });
-        getAllFormatterNames().forEach(function(formatterName) {
-            if (readmeText.indexOf(formatterName) === -1) {
-                grunt.fail.warn('A formatter was found that is not documented in README.md: ' + formatterName);
+            var readmeText = grunt.file.read('README.md', { encoding: 'UTF-8' });
```
Comment (ipip2005) [inline Gruntfile.js]: revert changes of indent

File: Gruntfile.js [hunk 3]
```diff
@@ -293,26 +298,26 @@ module.exports = function(grunt) {
...
+            });
+
+            if (readmeText.indexOf('\nVersion ' + packageJson.version + ' ') === -1) {
+                grunt.fail.warn('Version not documented in README.md correctly.\n' +
+                    'package.json declares: ' + packageJson.version + '\n' +
+                    'README.md declares something different.');
             }
         });
 
-        if (readmeText.indexOf('\nVersion ' + packageJson.version + ' ') === -1) {
```
Comment (ipip2005) [inline Gruntfile.js]: revert changes of indent

File: src/utils/implicitRoles/index.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+import { SECTION } from './section';
+import { SELECT } from './select';
+import { TBODY } from './tbody';
+import { TEXTAREA } from './textarea';
+import { TFOOT } from './tfoot';
+import { THEAD } from './thead';
+import { UL } from './ul';
+
+/**
+ * All functions of getting implicit role.
```
Comment (ipip2005) [inline src/utils/implicitRoles/index.ts]: Export function for getting implicit role based on tag name.

File: tslint.json [hunk 5]
```diff
@@ -1,5 +1,11 @@
 {
   "rules": {
+    "a11y-img-has-alt": true,
+    "a11y-props": true,
+    "a11y-role-has-required-aria-props": true,
+    "a11y-role": true,
+    "a11y-role-supports-aria-props": true,
+    "a11y-tabindex-no-positive": true,
```
Comment (ipip2005) [inline tslint.json]: Are these rules supposed to be enabled by default? @HamletDRC

File: src/a11yImgHasAltRule.ts [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+}
+
+export function getFailureStringEmptyAlt(tagName: string): string {
+  return `The value of 'alt' attribute in <${tagName}> tag is undefined or empty. \
+Add more details in 'alt' attribute or use role='presentation' for presentational images. \
+A reference for the presentation role can be found at https://www.w3.org/TR/wai-aria/roles#presentation.`;
+}
+
+export class Rule extends Lint.Rules.AbstractRule {
+  public static metadata: ExtendedMetadata = {
```
Comment (t-ligu) [inline src/a11yImgHasAltRule.ts]: How to code metadata? @HamletDRC

**Comment by HamletDRC** [comment]
Taking a look now...

**Comment by HamletDRC** [comment]
@t-ligu GitHub is asking you to sign the "cla" because you have not linked your GitHub account to your Microsoft email account. I think you can do either of the following: 
* Click "Agree" on the CLA request from GitHub
* Go to https://opensourcehub.microsoft.com/ (from CORPNET) and click the little red chain icon in the upper right. Basically, you will click "OK" a few times and then authenticate your GitHub account to the Microsoft GitHub org. Then you appear here as part of the official Microsoft team :)

**Comment by t-ligu** [comment]
@HamletDRC I solved the cla-required issue. Thanks.

**Pull request closed without merging**